File: Comuns/Diffle-Helman.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Comuns
{
    public static class Diffle_Helman
    {
        public static int G = 17;
        public static int N = 127;

        public static UInt64 CalculateR(int privateNumber) => Convert.ToUInt64(Math.Pow(G, privateNumber) % N);
        public static UInt64 CalculateK(int privateNumber, UInt64 R) => Convert.ToUInt64(Math.Pow(R, privateNumber) % N);
    }
}

File: Python Implementation/diffle_helman.py
class DiffleHelman:
    G = 7
    N = 23

    @staticmethod
    def calculate_r(private_number):
        return pow(DiffleHelman.G, private_number, DiffleHelman.N)

    @staticmethod
    def calculate_k(private_number, R):
        return pow(R, private_number, DiffleHelman.N)